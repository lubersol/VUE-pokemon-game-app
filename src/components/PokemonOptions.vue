<template>
  <div class="options-container">
    <ul class="options-list">
      <li
        v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click="$emit('selectionPokemon', pokemon.id)"
      >
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
  },
  methods: {},
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
/* Pokemon Options */
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: #b5e7a0;
}

.options-container {
  display: flex;
  justify-content: center;
  align-items: center;
}
.options-list {
  align-items: center;
  justify-content: center;
  padding: 0;
}
</style>
