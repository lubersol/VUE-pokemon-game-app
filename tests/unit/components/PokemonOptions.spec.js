import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'

import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', ()=>{

    let wrapper

    beforeEach(()=>{
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons
            }
        })
    })

    test('debe de hacer match con el snapshot', ()=>{

        console.log(wrapper.html())
        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('debe de mostrar las 4 opciones correctamente', () => {

        const liTags = wrapper.findAll('li')
        expect( liTags.length ).toBe(4)

        expect( liTags[0].text() ).toBe('pikachu')
        expect( liTags[1].text() ).toBe('charmander')
        expect( liTags[2].text() ).toBe('venusaur')
        expect( liTags[3].text() ).toBe('mew')

    })

    test('debe de emitir "selectionPokemon" con sus respectivos parametros al hacer click', ()=>{

        const [ li1, li2, li3, li4 ] = wrapper.findAll('li')

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect(wrapper.emitted('selectionPokemon').length).toBe(4)
        expect(wrapper.emitted('selectionPokemon')[0]).toEqual([5])
        expect(wrapper.emitted('selectionPokemon')[1]).toEqual([10])
        expect(wrapper.emitted('selectionPokemon')[2]).toEqual([15])
        expect(wrapper.emitted('selectionPokemon')[3]).toEqual([20])

    })

})