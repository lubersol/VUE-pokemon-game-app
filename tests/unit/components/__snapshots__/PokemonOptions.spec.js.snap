// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul class="options-list">
    <li>pikachu</li>
    <li>charmander</li>
    <li>venusaur</li>
    <li>mew</li>
  </ul>
</div>
`;
